### DISCUSSION TITLE
How can we export the opentelemetry metrics to traces spans?

### DISCUSSION THREAD

**Discussion by chandru-kumar** [open-telemetry/opentelemetry-java#3148] [Q&A]
How can we export the opentelemetry metrics to traces spans? I see metrics example code is instrumenting jvm memory, but not using anywhere like attaching it to Span attributes or exporting to traces. How can we use OpenTelemetry metrics to attach it to Span? 

Couldnt understand the usage of metrics with the examples..Can you pls provide us some good exmaples which shows end to end usage of metrics APIs?

**Comment by jkwatson** [comment]
I'm going to convert this to a discussion, since it's a question, not a bug.

**Comment by jkwatson** [comment]
Can you explain what you mean by "export metrics to spans"? Can you give an example of what you're asking?

**Reply by chandru-kumar** [reply]
So..When we get metrics like memory, cpu usage using metrics API, I'm thinking of to add those metrics to the span using span attributes.. And then export the trace uisng zipkin or jaeger..

From the examples given in metrics, I see the Instruments objetcs like Counter, Recorder and Observer just collects the metrics and not using further..So we as a user didnt get the exact usage of metrics APIs and dont know how to export the metrics which we collected using metrics API..

**Reply by jkwatson** [reply]
There isn't any built-in or recommended way to do this at the moment. It would definitely involve doing some custom work with a custom SpanProcessor that is somehow metrics-aware. If you figure out a way to do this, please feel free to let us know, and contribute a piece of example code that demonstrates it.

**Reply by chandru-kumar** [reply]
Sure..Thanks

**Reply by iNikem** [reply]
Just to make sure we are on the same page, @ChandrewR . Usually metrics are not meant to be exported as span attributes. You should export metrics as metrics, using MetricExporter, to metric backend, such as e.g. Prometheus. Traces and Metrics are sufficiently different and thus are viewed separately in OpenTelemetry.

**Reply by chandru-kumar** [reply]
We were thinking in the aspects of getting applications metrics like jvm memory etc when we do an operation in application..We thought to include a span for an operation and get the resource usage like application's metrics and add it to span..

Ok..Do we have any examples for Metric Exporter? If not so, can you pls include one in examples? It will be helpful and much appreciated..

**Reply by jkwatson** [reply]
The otlp example includes using the metric exporter. As does the prometheus example. Is there something else you were looking for?

**Reply by chandru-kumar** [reply]
Ok..I will look into those examples. Thank you!

**Reply by itsprakash84** [reply]
Hi,

I am looking for Prometheus example, I want to know what are metric by default exported using automatic instrumentation?

I assume we no need to write any code to get Metrics, because in spring boot we just define the below properties rest all taken care. 
management.endpoints.web.exposure.include=prometheus

Metrics like 

1) 
# HELP http_server_requests_seconds  
# TYPE http_server_requests_seconds summary
http_server_requests_seconds_count{exception="None",method="GET",outcome="SUCCESS",status="200",uri="/info",} 5.0
http_server_requests_seconds_sum{exception="None",method="GET",outcome="SUCCESS",status="200",uri="/info",} 0.0294437

2)
http_server_requests_seconds_max{exception="None",method="GET",outcome="SUCCESS",status="200",uri="/info",} 0.0
http_server_requests_seconds_max{exception="None",method="GET",outcome="SUCCESS",status="200",uri="/metrics",} 0.0048482

3)
jvm_memory_used_bytes

4)
system_cpu_usage 0.0019443244730260306
 5)
Tomcat metrics
tomcat_threads_busy_threads

**Reply by jkwatson** [reply]
I don't know that the auto-instrumentation provides any metrics at this point. That's a question for the opentelemetry-java-instrumentation project, though.
